pgjdbc/src/test/java/org/postgresql/util/OSUtilTest.java

```diff
@@ -27,7 +27,7 @@ void getUserConfigRootDirectory() throws Exception {
         new SystemProperties("os.name", "Windows 10")).execute(() -> {
           String result = OSUtil.getUserConfigRootDirectory();
           assertNotNull(result);
```

Comment:
would you please remove `assertNotNull`? It adds no value, and the later `assertEquals` would handle nulls properly

Reply:
In this case, it's true, but they are not actually the same, the `assertNotNull()` makes it explicit that a non-null value is expected, while the `assertEquals()` would mean that the expected value is not equal to the returned null.

This is even more important to distinguish when you call a method from the returned object, using something like 
```java
assertEquals("x", result.trim());
```
as that would throw an NPE.

In general, I prefer explicit checks even if they are somewhat redundant, but that is just a personal taste.

Reply:
@jorsol , when you add `assertions`, please think how **the failure** would look like.

The failure should look like a **good** bug report.

Now let's consider: `assertNotNull(result);`

The failure would look like:

```
  AssertionError: expected non null value, got null
  at ....
```

Do you think this is a good bug report?
I am sure this is awful

Reply:
I agree, but using an explicit message would solve how the failure would look like.

Anyway, in this case, yes, it's probably useless as the String is used directly without calling any method.

But as you mention, the failure should look like a good bug report, and if the object calls a method (wich is not the case here) you would get an awful `java.lang.NullPointerException` and that's even worse than the `AssertionFailedError: expected: not <null>`.

Reply:
>  if the object calls a method (wich is not the case here) you

Let us discuss it when concrete examples appear.
`assertNotNull` could work as a standalone, single assertion in a test.
However, if there's a more specific assertion, then `assertNotNull` often adds no value.